pkg/upgrade/requestor/upgrade_requestor.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2025 NVIDIA CORPORATION & AFFILIATES
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package requestor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	//nolint:depguard
+	maintenancev1alpha1 "github.com/Mellanox/maintenance-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/NVIDIA/k8s-operator-libs/api/upgrade/v1alpha1"
+	"github.com/NVIDIA/k8s-operator-libs/pkg/consts"
+	"github.com/NVIDIA/k8s-operator-libs/pkg/upgrade/base"
+)
+
+const (
+	MaintenanceOPFinalizerName      = "maintenance.nvidia.com/finalizer"
+	MaintenanceOPDefaultRequestorID = "nvidia.network.operator"
+	MaintenanceOPEvictionGPU        = "nvidia.com/gpu-*"
+	MaintenanceOPEvictionRDMA       = "nvidia.com/rdma*"
```

Comment:
Can we add `ResourceName` suffix to the variables' names? It's a little confusing what they are supposed to mean